src/ODBC/OdbcConnection.php

```diff
@@ -64,12 +64,15 @@ protected function connect(): void
             /** @var resource|false $connection */
             $connection = @odbc_connect($this->dsn, $this->user, $this->password, $this->odbcCursorMode);
         } catch (Throwable $e) {
+            $this->handleConnectionError($e->getMessage(), $e->getCode(), $e);
             throw new OdbcException($e->getMessage(), $e->getCode(), $e);
         }
 
         // "odbc_connect" can generate warning, if "set_error_handler" is not set, so we are checking it manually
         if ($connection === false) {
-            throw new OdbcException(odbc_errormsg() . ' ' . odbc_error());
+            $message = odbc_errormsg() . ' ' . odbc_error();
+            $this->handleConnectionError($message);
+            throw new OdbcException($message);
```

Comment:
nechtěl jsi dát ten throw dovnitř handleConnectionError? :)

Reply:
no bylo by to fajn, ale když je connection na false nebo se vyhodí vyjímka tak chci aby to **vždycky** spadlo a když to dám dovnitř a v nějakém extractoru to přepíšu tak to spadnout nemusí

Reply:
ok